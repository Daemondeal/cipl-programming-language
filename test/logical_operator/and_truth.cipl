# Only false and null should be not truthy
print(false and "bad")
print(null and "bad")

print(true and "ok")
print(0 and "ok")
print("" and "ok")