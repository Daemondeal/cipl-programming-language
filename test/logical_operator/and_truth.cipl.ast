(expression (call print (and false bad)))
(expression (call print (and null bad)))
(expression (call print (and true ok)))
(expression (call print (and 0 ok)))
(expression (call print (and  ok)))

