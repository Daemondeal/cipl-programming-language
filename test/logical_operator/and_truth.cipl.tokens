Token          Lexeme         Line 
IDENTIFIER     println        2    
LEFT_PAREN     (              2    
FALSE          false          2    
AND            and            2    
STRING         "bad"          2     bad
RIGHT_PAREN    )              2    
LINE_END       \n             3    
IDENTIFIER     println        3    
LEFT_PAREN     (              3    
NULL           null           3    
AND            and            3    
STRING         "bad"          3     bad
RIGHT_PAREN    )              3    
LINE_END       \n             4    
IDENTIFIER     println        5    
LEFT_PAREN     (              5    
TRUE           true           5    
AND            and            5    
STRING         "ok"           5     ok
RIGHT_PAREN    )              5    
LINE_END       \n             6    
IDENTIFIER     println        6    
LEFT_PAREN     (              6    
NUMBER         0              6     0
AND            and            6    
STRING         "ok"           6     ok
RIGHT_PAREN    )              6    
LINE_END       \n             7    
IDENTIFIER     println        7    
LEFT_PAREN     (              7    
STRING         ""             7     
AND            and            7    
STRING         "ok"           7     ok
RIGHT_PAREN    )              7    
LINE_END       \n             7    
EOF                           7    
