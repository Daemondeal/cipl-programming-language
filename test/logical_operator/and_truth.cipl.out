::stdout::
false
null
ok
ok
ok

::stderr::
